src/Illuminate/Routing/Router.php

```diff
@@ -702,11 +701,14 @@ protected function runRouteWithinStack(Route $route, Request $request)
 	 */
 	public function gatherRouteMiddlewares(Route $route)
 	{
-		return Collection::make($route->middleware())->map(function($m)
+		$result = [];
+
+		foreach ($route->middleware() as $tag)
 		{
-			return Collection::make(array_get($this->middleware, $m, $m));
+			$result[] = (isset($this->middleware[$tag]) === true ? $this->middleware[$tag] : $tag);
```

Comment:
Do we need to explicitly compare to `true`, knowing that `isset()` returns `BOOL`?